### PR TITLE
3251 Add dependency check in WSIReader

### LINKED ISSUE #3251: `tifffile` support in WSIReader
**Is your feature request related to a problem? Please describe.**
WSIReader currently supports `cuCIM` and `OpenSlide`, which can read patches from very large whole slide images without loading the entire image into memory. However, `tifffile` is another popular library to load TIFF files (which the most common format for whole slide imaging). Although it does not support extraction without loading the whole image, there are use cases that loading the whole slide image is desirable and `tifffile` can do it faster than `cuCIM` and `OpenSlide`.Thus supporting `tifffile` in `WSIReader` would be helpful.

Related issue: https://github.com/Project-MONAI/MONAI/issues/3161

### PR DESCRIPTION
**Pull request by Nic-Ma**
Fixes #3251 .

### Description
This PR is follow up of #3251 to add the dependency check in WSIReader.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Review by yiheng-wang-nv — APPROVED** [review]

File: monai/data/image_reader.py [hunk 1]
```diff
@@ -690,14 +694,18 @@ class WSIReader(ImageReader):
...
-        if self.backend == "openslide":
-            self.wsi_reader, *_ = optional_import("openslide", name="OpenSlide")
-        elif self.backend == "cucim":
-            self.wsi_reader, *_ = optional_import("cucim", name="CuImage")
-        elif self.backend == "tifffile":
-            self.wsi_reader, *_ = optional_import("tifffile", name="TiffFile")
-        else:
-            raise ValueError('`backend` should be "cuCIM", "OpenSlide", or "TiffFile')
+
+        @require_pkg(pkg_name=self.backend)
```
Comment (wyli) [inline monai/data/image_reader.py:698]: Why hiding this API, we lose the flexibility of inheriting it easily

Comment (Nic-Ma) [reply monai/data/image_reader.py:698]: Hi @wyli ,

I think it's not supported in python to set `pkg_name=self.backend` directly if not hiding the function in `init()`:
```py
def __init__(self, backend: str = "OpenSlide", level: int = 0):
    super().__init__()
    self.backend = backend.lower()
    self.wsi_reader = self._set_reader(self.backend)
    self.level = level

@require_pkg(pkg_name=self.backend)
def _set_reader(self, backend: str):
    if backend == "openslide":
        return OpenSlide
    if backend == "cucim":
        return CuImage
    if backend == "tifffile":
        return TiffFile
    raise ValueError("`backend` should be 'cuCIM', 'OpenSlide' or 'TiffFile'.")
```

Maybe I missed some python feature?

Thanks.

Comment (wyli) [reply monai/data/image_reader.py:698]: I think you can use the decorator in the constructor as a normal function "require_pkg(backend)(self._set_reader)"

Comment (Nic-Ma) [reply monai/data/image_reader.py]: updated in the latest commit.

Thanks.

**Review by wyli — APPROVED** [review]
Thanks, it looks good to me.

**Pull request merged**